core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -986,7 +986,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     union(Seq(first) ++ rest)
 
   /** Get an RDD that has no partitions or elements. */
-  def emptyRDD[T: ClassTag] = new EmptyRDD[T](this)
+  def emptyRDD[T: ClassTag]: EmptyRDD[T] = new EmptyRDD[T](this)
```

Comment:
ah right, thanks for clarifying.

its really a shame that the scala compiler didn't warn in the first place that this method led to `EmptyRDD` was escaping its declared scope.